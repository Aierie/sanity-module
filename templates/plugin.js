import Vue from 'vue'

<%= options.client
  ? "import createClient from '@sanity/client'"
  : "import { createClient } from '@nuxtjs/sanity'"
%>

<% if (options.components.imageHelper) { %>
  import { SanityImage } from './sanity-image'
  Vue.component('SanityImage', SanityImage)
<% } %>

<% if (options.components.contentHelper) { %>
  import { SanityContent } from '@nuxtjs/sanity/dist/sanity-content'
  Vue.component('SanityContent', SanityContent)
<% } %>

const options = JSON.parse('<%= options.sanityConfig %>')

/**
 * @type {import('@nuxt/types').Plugin}
 */
export default async (_ctx, inject) => {
  inject('sanity', {
    client: createClient(options),
    fetch(...args) {
      return this.client.fetch(...args)
    },
    setToken(token) {
      this.client = createClient({ ...options, token })
    }
  })
}
