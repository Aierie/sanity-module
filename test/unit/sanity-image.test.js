import { mount } from '@vue/test-utils'

import { SanityImage } from '../../templates/sanity-image'

const getWrapper = propsData => mount(SanityImage, { propsData })

describe('SanityImage', () => {
  test('it parses asset IDs correctly', () => {
    const wrapper = getWrapper({
      assetId: 'image-7aa06723bb01a7a79055b6d6f5be80329a0e5b58-780x1170-jpg'
    })

    expect(wrapper.attributes().src).toBe(
      'https://cdn.sanity.io/images/<%= options.projectId %>/<%= options.dataset %>/7aa06723bb01a7a79055b6d6f5be80329a0e5b58-780x1170.jpg'
    )
  })

  test('it correctly adds query params', () => {
    const wrapper = getWrapper({
      assetId: 'image-7aa06723bb01a7a79055b6d6f5be80329a0e5b58-780x1170-jpg',
      w: 20,
      h: '21'
    })

    expect(wrapper.attributes().src).toBe(
      'https://cdn.sanity.io/images/<%= options.projectId %>/<%= options.dataset %>/7aa06723bb01a7a79055b6d6f5be80329a0e5b58-780x1170.jpg?h=21&w=20'
    )
  })
})
