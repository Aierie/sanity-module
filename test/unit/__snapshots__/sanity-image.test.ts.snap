// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SanityImage provides a valid renderless component 1`] = `<img src="https://cdn.sanity.io/images/undefined/production/G3i4emG6B8JnTmGoN0UjgAp8-300x450.jpg?auto=format">`;
